tests/Application.Tests/Users/UserServiceTests.cs

```diff
@@ -99,55 +99,71 @@ public void FetchAllUsers_ReturnsAllUsersInTheDatabase()
     }
 
     [Fact]
-    public void UserExists_ReturnsTrue_IfUserExistsInRepository()
+    public void ValidatePassword_shouldBeTrue_WhenPasswordIsUserPassword()
     {
         // Arrange
-        string username = "user";
-        _mockRepo.Setup(repository => repository.Exists(username)).Returns(true);
+        const string password = "UserPassword";
+        string passwordHash = BCrypt.Net.BCrypt.HashPassword(password);
+        var user = new User() { Password = passwordHash, };
+        _mockRepo.Setup(repository => repository.Fetch(user.Name)).Returns(user);
 
         // Act
-        bool actual = _cut.UserExists(username);
+        AuthenticateResponse authResponse = _cut.Authenticate(user.Name, password);
 
         // Assert
-        actual.Should().BeTrue();
+        authResponse.Should().NotBeNull();
     }
 
     [Fact]
-    public void UserExists_ReturnsFalse_IfUserDoesNotExistInRepository()
+    public void ValidatePassword_shouldBeFalse_WhenPasswordIsNotUserPassword()
+    {
+        // Arrange
+        const string password1 = "User1Password";
+        const string password2 = "User2Password";
```

Comment:
nit:
To make the test a bit more easier to understand you could be more specific:
```suggestion
        const string correctPassword = "CorrectPassword";
        const string incorrectPassword = "IncorrectPassword";
```